src/run_sign.py

```diff
@@ -32,22 +35,20 @@ def main():
     )
     args = parser.parse_args()
 
-    console.configure(level=args.logging_level)
-
-    manifest = BuildManifest.from_file(args.manifest)
-    basepath = os.path.dirname(os.path.abspath(args.manifest.name))
-    signer = Signer()
+    if args.sigType not in ACCEPTED_SIGNATURE_FILE_TYPES:
+        raise Exception(f"Accepted types of signature files are ${ACCEPTED_SIGNATURE_FILE_TYPES}")
 
-    for component in manifest.components.select(focus=args.component):
-        logging.info(f"Signing {component.name}")
-
-        for artifact_type in component.artifacts:
-            if args.type and args.type != artifact_type:
-                continue
+    console.configure(level=args.logging_level)
 
-            signer.sign_artifacts(component.artifacts[artifact_type], basepath)
+    file_extension = args.target.suffix
 
-    logging.info("Done.")
+    if file_extension == ".yml":
```

Comment:
Add a class called Signers that has a method called from_file that returns either SignWithManifest or SignArtifacts. Both should inherit from the same base class and have a .sign method that you call once. The duplicate logic of creating a Signer should also be refactored - I think you want to get a list of files to sign in different ways but then sign each item in the list in the base class.